src/rdgz.c

```diff
@@ -26,15 +26,16 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
+#include "rdkafka_int.h"
+
 #include "rd.h"
 #include "rdgz.h"
 
 #include <zlib.h>
 
-
 #define RD_GZ_CHUNK  262144
 
-void *rd_gz_decompress (const void *compressed, int compressed_len,
+void *rd_gz_decompress (rd_kafka_broker_t *rkb, const void *compressed, int compressed_len,
```

Comment:
This is intertwining abstractions.
Better to pass an errstr and errstr_size to rd_gz_decompress where it can write an error string which can then be logged from the caller.